packages/next-swc/crates/next-core/src/next_app/mod.rs

```diff
@@ -6,12 +6,283 @@ pub(crate) mod app_page_entry;
 pub(crate) mod app_route_entry;
 pub(crate) mod unsupported_dynamic_metadata_issue;
 
-pub use app_client_references_chunks::{
-    get_app_client_references_chunks, ClientReferenceChunks, ClientReferencesChunks,
+use std::{
+    fmt::{Display, Formatter, Write},
+    ops::Deref,
 };
-pub use app_client_shared_chunks::get_app_client_shared_chunks;
-pub use app_entry::AppEntry;
-pub use app_favicon_entry::get_app_route_favicon_entry;
-pub use app_page_entry::get_app_page_entry;
-pub use app_route_entry::get_app_route_entry;
-pub use unsupported_dynamic_metadata_issue::UnsupportedDynamicMetadataIssue;
+
+use anyhow::{bail, Result};
+use serde::{Deserialize, Serialize};
+use turbo_tasks::{trace::TraceRawVcs, TaskInput};
+
+pub use crate::next_app::{
+    app_client_references_chunks::{
+        get_app_client_references_chunks, ClientReferenceChunks, ClientReferencesChunks,
+    },
+    app_client_shared_chunks::get_app_client_shared_chunks,
+    app_entry::AppEntry,
+    app_favicon_entry::get_app_route_favicon_entry,
+    app_page_entry::get_app_page_entry,
+    app_route_entry::get_app_route_entry,
+    unsupported_dynamic_metadata_issue::UnsupportedDynamicMetadataIssue,
+};
+
+#[derive(Clone, Debug, Hash, Serialize, Deserialize, PartialEq, Eq, TaskInput, TraceRawVcs)]
+pub enum PageSegment {
+    Static(String),
+    Dynamic(String),
+    CatchAll(String),
+    OptionalCatchAll(String),
+    Group(String),
+    Parallel(String),
+    PageType(PageType),
```

Comment:
Where is this used? I don't see it created anywhere.

Reply:
it's not used yet, it should replace this in the future:
https://github.com/vercel/next.js/blob/6803c303784233fadedf3e499a187511b588e502/packages/next-swc/crates/next-core/src/next_app/app_page_entry.rs#L155 (there's also the equivalent in `app_route_entry.rs`)